### PR TITLE
FRDQ-59 - Create nginx_deploy_aws.yml

### PR DESCRIPTION
**Pull request by hayuna**

### CODE REVIEW

**Comment by hayuna** [comment]
<p>Total Coverage: <code>63.33</code></p>
<details><summary>Coverage report</summary>
<p>
<pre>
> frontend@0.1.0 test:coverage /home/runner/work/frontend/frontend
> npm test -- --coverage


> frontend@0.1.0 test /home/runner/work/frontend/frontend
> CI=true react-scripts test --env=jsdom "--coverage"

------------------------------------------|----------|----------|----------|----------|-------------------|
File                                      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
------------------------------------------|----------|----------|----------|----------|-------------------|
All files                                 |    63.33 |    85.71 |       50 |    63.33 |                   |
 src                                      |        0 |      100 |        0 |        0 |                   |
  App.tsx                                 |        0 |      100 |        0 |        0 |                 4 |
  index.tsx                               |        0 |      100 |      100 |        0 |                 7 |
  react-app-env.d.ts                      |        0 |        0 |        0 |        0 |                   |
 src/components/ExampleStorybookComponent |        0 |      100 |        0 |        0 |                   |
  ExampleStorybookComponent.model.ts      |        0 |        0 |        0 |        0 |                   |
  ExampleStorybookComponent.tsx           |        0 |      100 |        0 |        0 |              9,10 |
 src/components/atoms/Button              |    90.91 |    90.91 |      100 |    90.91 |                   |
  Button.model.ts                         |        0 |        0 |        0 |        0 |                   |
  Button.tsx                              |    90.91 |    90.91 |      100 |    90.91 |                51 |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Header              |      100 |      100 |      100 |      100 |                   |
  Header.tsx                              |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/List                |        0 |      100 |        0 |        0 |                   |
  List.model.ts                           |        0 |        0 |        0 |        0 |                   |
  List.tsx                                |        0 |      100 |        0 |        0 |          12,13,16 |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/ListItem            |        0 |        0 |        0 |        0 |                   |
  ListItem.model.ts                       |        0 |        0 |        0 |        0 |                   |
  ListItem.tsx                            |        0 |        0 |        0 |        0 |             11,12 |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/SnackBar            |      100 |      100 |      100 |      100 |                   |
  SnackBar.tsx                            |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/SocialButton        |      100 |      100 |      100 |      100 |                   |
  SocialButton.model.ts                   |        0 |        0 |        0 |        0 |                   |
  SocialButton.tsx                        |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/styles                               |       75 |      100 |      100 |       75 |                   |
  GlobalStyles.tsx                        |        0 |      100 |      100 |        0 |                 4 |
  colors.ts                               |      100 |      100 |      100 |      100 |                   |
  devices.ts                              |      100 |      100 |      100 |      100 |                   |
  typography.ts                           |      100 |      100 |      100 |      100 |                   |
------------------------------------------|----------|----------|----------|----------|-------------------|
</pre>
</p>
</details>

**Comment by bwieckow** [comment]
Name of the file should be `frontend_deploy_aws.yml`

File: .github/workflows/nginx_deploy_aws.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    - name: Login to Amazon ECR
+      id: login-ecr
+      uses: aws-actions/amazon-ecr-login@v1
+
+    - name: Build, tag, and push image to Amazon ECR
+      id: build-image
+      env:
+        ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        ECR_REPOSITORY: nginx
```
Comment (bwieckow) [inline .github/workflows/nginx_deploy_aws.yml:59]: ECR_REPOSITORY: frontend

File: .github/workflows/nginx_deploy_aws.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        # be deployed to ECS.
+        cd ./nginx
+        docker build -t $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG .
+        docker push $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
+        echo "::set-output name=image::$ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG"
+    - name: Fill in the new image ID in the Amazon ECS task definition
+      id: task-def
+      uses: aws-actions/amazon-ecs-render-task-definition@v1
+      with:
+        task-definition: ./nginx/nginx_task_definition.json
```
Comment (bwieckow) [inline .github/workflows/nginx_deploy_aws.yml:73]: It is recommanded to place `frontend_deploy_aws.yml` in repository root directory 
This should be:
`task-definition: ./frontned_task_definition.json`

File: .github/workflows/nginx_deploy_aws.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        container-name: nginx
```
Comment (bwieckow) [inline .github/workflows/nginx_deploy_aws.yml:74]: frontend

File: .github/workflows/nginx_deploy_aws.yml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        image: ${{ steps.build-image.outputs.image }}
+
+    - name: Deploy Amazon ECS task definition
+      uses: aws-actions/amazon-ecs-deploy-task-definition@v1
+      with:
+        task-definition: ${{ steps.task-def.outputs.task-definition }}
+        service: nginx
```
Comment (bwieckow) [inline .github/workflows/nginx_deploy_aws.yml:81]: frontend

File: .github/workflows/nginx_deploy_aws.yml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    - name: Stop current task
+      run: |
+        #!/bin/bash
+        tasks=`aws ecs list-tasks --cluster ProgrammersOnly | jq '.taskArns'`
+        for task in $tasks; do
+            name=""
+            if [ ${#task} -gt 1 ]; then
+                task=`echo $task | tr -d '\"'`
+                name=`aws ecs describe-tasks --cluster ProgrammersOnly --tasks $task | jq '.tasks' | jq '.[0].overrides.containerOverrides' | jq '.[0].name' | tr -d '\"'`
+                if [[ $name == "nginx" ]]; then
```
Comment (bwieckow) [inline .github/workflows/nginx_deploy_aws.yml:95]: `$name == "frontend"`

**Review by bwieckow — CHANGES_REQUESTED** [review]
Change filename to `frontend_aws_deploy.yml`

File: .github/workflows/nginx_deploy_aws.yml (hunk 2 above)
Comment (hayuna) [reply .github/workflows/nginx_deploy_aws.yml:73]: @bwieckow should I move this file in root directory instead of current place?

**Comment by hayuna** [comment]
<p>Total Coverage: <code>63.33</code></p>
<details><summary>Coverage report</summary>
<p>
<pre>
> frontend@0.1.0 test:coverage /home/runner/work/frontend/frontend
> npm test -- --coverage


> frontend@0.1.0 test /home/runner/work/frontend/frontend
> CI=true react-scripts test --env=jsdom "--coverage"

------------------------------------------|----------|----------|----------|----------|-------------------|
File                                      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
------------------------------------------|----------|----------|----------|----------|-------------------|
All files                                 |    63.33 |    85.71 |       50 |    63.33 |                   |
 src                                      |        0 |      100 |        0 |        0 |                   |
  App.tsx                                 |        0 |      100 |        0 |        0 |                 4 |
  index.tsx                               |        0 |      100 |      100 |        0 |                 7 |
  react-app-env.d.ts                      |        0 |        0 |        0 |        0 |                   |
 src/components/ExampleStorybookComponent |        0 |      100 |        0 |        0 |                   |
  ExampleStorybookComponent.model.ts      |        0 |        0 |        0 |        0 |                   |
  ExampleStorybookComponent.tsx           |        0 |      100 |        0 |        0 |              9,10 |
 src/components/atoms/Button              |    90.91 |    90.91 |      100 |    90.91 |                   |
  Button.model.ts                         |        0 |        0 |        0 |        0 |                   |
  Button.tsx                              |    90.91 |    90.91 |      100 |    90.91 |                51 |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Header              |      100 |      100 |      100 |      100 |                   |
  Header.tsx                              |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/List                |        0 |      100 |        0 |        0 |                   |
  List.model.ts                           |        0 |        0 |        0 |        0 |                   |
  List.tsx                                |        0 |      100 |        0 |        0 |          12,13,16 |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/ListItem            |        0 |        0 |        0 |        0 |                   |
  ListItem.model.ts                       |        0 |        0 |        0 |        0 |                   |
  ListItem.tsx                            |        0 |        0 |        0 |        0 |             11,12 |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/SnackBar            |      100 |      100 |      100 |      100 |                   |
  SnackBar.tsx                            |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/SocialButton        |      100 |      100 |      100 |      100 |                   |
  SocialButton.model.ts                   |        0 |        0 |        0 |        0 |                   |
  SocialButton.tsx                        |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/styles                               |       75 |      100 |      100 |       75 |                   |
  GlobalStyles.tsx                        |        0 |      100 |      100 |        0 |                 4 |
  colors.ts                               |      100 |      100 |      100 |      100 |                   |
  devices.ts                              |      100 |      100 |      100 |      100 |                   |
  typography.ts                           |      100 |      100 |      100 |      100 |                   |
------------------------------------------|----------|----------|----------|----------|-------------------|
</pre>
</p>
</details>

**Comment by hayuna** [comment]
<p>Total Coverage: <code>63.33</code></p>
<details><summary>Coverage report</summary>
<p>
<pre>
> frontend@0.1.0 test:coverage /home/runner/work/frontend/frontend
> npm test -- --coverage


> frontend@0.1.0 test /home/runner/work/frontend/frontend
> CI=true react-scripts test --env=jsdom "--coverage"

------------------------------------------|----------|----------|----------|----------|-------------------|
File                                      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
------------------------------------------|----------|----------|----------|----------|-------------------|
All files                                 |    63.33 |    85.71 |       50 |    63.33 |                   |
 src                                      |        0 |      100 |        0 |        0 |                   |
  App.tsx                                 |        0 |      100 |        0 |        0 |                 4 |
  index.tsx                               |        0 |      100 |      100 |        0 |                 7 |
  react-app-env.d.ts                      |        0 |        0 |        0 |        0 |                   |
 src/components/ExampleStorybookComponent |        0 |      100 |        0 |        0 |                   |
  ExampleStorybookComponent.model.ts      |        0 |        0 |        0 |        0 |                   |
  ExampleStorybookComponent.tsx           |        0 |      100 |        0 |        0 |              9,10 |
 src/components/atoms/Button              |    90.91 |    90.91 |      100 |    90.91 |                   |
  Button.model.ts                         |        0 |        0 |        0 |        0 |                   |
  Button.tsx                              |    90.91 |    90.91 |      100 |    90.91 |                51 |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Header              |      100 |      100 |      100 |      100 |                   |
  Header.tsx                              |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/List                |        0 |      100 |        0 |        0 |                   |
  List.model.ts                           |        0 |        0 |        0 |        0 |                   |
  List.tsx                                |        0 |      100 |        0 |        0 |          12,13,16 |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/ListItem            |        0 |        0 |        0 |        0 |                   |
  ListItem.model.ts                       |        0 |        0 |        0 |        0 |                   |
  ListItem.tsx                            |        0 |        0 |        0 |        0 |             11,12 |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/SnackBar            |      100 |      100 |      100 |      100 |                   |
  SnackBar.tsx                            |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/SocialButton        |      100 |      100 |      100 |      100 |                   |
  SocialButton.model.ts                   |        0 |        0 |        0 |        0 |                   |
  SocialButton.tsx                        |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/styles                               |       75 |      100 |      100 |       75 |                   |
  GlobalStyles.tsx                        |        0 |      100 |      100 |        0 |                 4 |
  colors.ts                               |      100 |      100 |      100 |      100 |                   |
  devices.ts                              |      100 |      100 |      100 |      100 |                   |
  typography.ts                           |      100 |      100 |      100 |      100 |                   |
------------------------------------------|----------|----------|----------|----------|-------------------|
</pre>
</p>
</details>

File: .github/workflows/nginx_deploy_aws.yml (hunk 2 above)
Comment (bwieckow) [reply .github/workflows/nginx_deploy_aws.yml]: @hayuna Nope, it is correct now. 

Only `Dockerfile` and `frontend_task_definition.json` should have been placed in repository root directory.

File: .github/workflows/nginx_deploy_aws.yml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+      id: build-image
+      env:
+        ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        ECR_REPOSITORY: frontend
+        IMAGE_TAG: ${{ github.sha }}
+      run: |
+        # Build a docker container and
+        # push it to ECR so that it can
+        # be deployed to ECS.
+        cd ./nginx
```
Comment (bwieckow) [inline .github/workflows/nginx_deploy_aws.yml:65]: To be deleted

**Review by bwieckow — CHANGES_REQUESTED** [review]

File: .github/workflows/nginx_deploy_aws.yml (hunk 6 above)
Comment (bwieckow) [reply .github/workflows/nginx_deploy_aws.yml:65]: And please change FILENAME from `nginx_deploy_aws.yml` to 'frontend_deploy_aws.yml`

**Comment by hayuna** [comment]
<p>Total Coverage: <code>63.33</code></p>
<details><summary>Coverage report</summary>
<p>
<pre>
> frontend@0.1.0 test:coverage /home/runner/work/frontend/frontend
> npm test -- --coverage


> frontend@0.1.0 test /home/runner/work/frontend/frontend
> CI=true react-scripts test --env=jsdom "--coverage"

------------------------------------------|----------|----------|----------|----------|-------------------|
File                                      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
------------------------------------------|----------|----------|----------|----------|-------------------|
All files                                 |    63.33 |    85.71 |       50 |    63.33 |                   |
 src                                      |        0 |      100 |        0 |        0 |                   |
  App.tsx                                 |        0 |      100 |        0 |        0 |                 4 |
  index.tsx                               |        0 |      100 |      100 |        0 |                 7 |
  react-app-env.d.ts                      |        0 |        0 |        0 |        0 |                   |
 src/components/ExampleStorybookComponent |        0 |      100 |        0 |        0 |                   |
  ExampleStorybookComponent.model.ts      |        0 |        0 |        0 |        0 |                   |
  ExampleStorybookComponent.tsx           |        0 |      100 |        0 |        0 |              9,10 |
 src/components/atoms/Button              |    90.91 |    90.91 |      100 |    90.91 |                   |
  Button.model.ts                         |        0 |        0 |        0 |        0 |                   |
  Button.tsx                              |    90.91 |    90.91 |      100 |    90.91 |                51 |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Header              |      100 |      100 |      100 |      100 |                   |
  Header.tsx                              |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/List                |        0 |      100 |        0 |        0 |                   |
  List.model.ts                           |        0 |        0 |        0 |        0 |                   |
  List.tsx                                |        0 |      100 |        0 |        0 |          12,13,16 |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/ListItem            |        0 |        0 |        0 |        0 |                   |
  ListItem.model.ts                       |        0 |        0 |        0 |        0 |                   |
  ListItem.tsx                            |        0 |        0 |        0 |        0 |             11,12 |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/SnackBar            |      100 |      100 |      100 |      100 |                   |
  SnackBar.tsx                            |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/SocialButton        |      100 |      100 |      100 |      100 |                   |
  SocialButton.model.ts                   |        0 |        0 |        0 |        0 |                   |
  SocialButton.tsx                        |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/styles                               |       75 |      100 |      100 |       75 |                   |
  GlobalStyles.tsx                        |        0 |      100 |      100 |        0 |                 4 |
  colors.ts                               |      100 |      100 |      100 |      100 |                   |
  devices.ts                              |      100 |      100 |      100 |      100 |                   |
  typography.ts                           |      100 |      100 |      100 |      100 |                   |
------------------------------------------|----------|----------|----------|----------|-------------------|
</pre>
</p>
</details>

**Comment by hayuna** [comment]
Waiting for dockerfile and task definition

**Review by bwieckow — APPROVED** [review]

**Pull request closed without merging**